pennylane/qchem/factorization.py

```diff
@@ -452,3 +456,105 @@ def _chemist_transform(one_body_tensor=None, two_body_tensor=None, spatial_basis
+    its spectral range, based on the number of electron operator :math:`\hat{N}_e`.
+    The shifted Hamiltonian (:math:`\hat{H}^{\prime}`) will be given by:
+
+    .. math::
+
+        H^{\prime}(k_1, k_2, \vec{\xi}) = \hat{H} - k_1 (\hat{N}_e - N_e) - k_2 (\hat{N}_e^2 - \hat{N}_e^2) + \sum_{ij}\xi_{ij} T_{ij} (\hat{N}_e - N_e),
+
+    where :math:`N_e` is the target number of electrons and :math:`\vec{\xi}` is
+    a real vector with symmetric elements.
+
+    Args:
+        one_electron (array[float]): a one-electron integral tensor giving the :math:`T_{pq}`.
+        two_electron (array[float]): a two-electron integral tensor in the chemist notation giving the :math:`V_{pqrs}`.
+        n_elec (bool): target number of electrons for selecting the target eigenstates.
+        method (str | callable): type of solver used by ``scipy.optimize.minimize``. Please refer to its
+            `documentation <https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize>`_
+            for the list of all available solvers. Default solver is `"L-BFGS-B"`
+
+    Returns:
+        tuple(array[float], array[float], array[float]): symmetry shifted core, one-body tensor and two-body tensor for the provided terms.
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = qml.numpy.array([[0.0, 0.0, 0.0], [1.398397361, 0.0, 0.0]], requires_grad=False)
+    >>> mol = qml.qchem.Molecule(symbols, geometry, basis_name="STO-3G")
+    >>> core, one, two = qml.qchem.electron_integrals(mol)()
+    >>> ctwo = np.swapaxes(two, 1, 3)
+    >>> shifted_core, shifted_one, shifted_two = symmetry_shift(core, one, ctwo, n_elec=mol.n_electrons)
+    >>> print(shifted_two)
+    [[[[ 1.12461110e-02 -1.70030746e-09]
+      [-1.70030746e-09 -1.12461660e-02]]
+     [[-1.70030746e-09  1.81210462e-01]
+      [ 1.81210462e-01 -1.70032620e-09]]]
+     [[[-1.70030763e-09  1.81210462e-01]
+      [ 1.81210462e-01 -1.70032598e-09]]
+     [[-1.12461660e-02 -1.70032620e-09]
+      [-1.70032620e-09  1.12461854e-02]]]]
+    """
+    norb = one_electron.shape[0]
+    ki_vec = np.array([0.0, 0.0])
+    xi_mat = np.zeros((norb, norb))
+    xi_idx = np.tril_indices_from(xi_mat)
+    xi_vec = xi_mat[xi_idx]
+
+    params = np.hstack((ki_vec, xi_vec))
+    cost_func = partial(
+        _symmetry_shift_two_body_loss, two=two_electron, xi_idx=xi_idx
+    )  # Step 1: Reduce norm of two-body term
+
+    res_two = sp.optimize.minimize(cost_func, params, method=method)
+    _, k2, xi, _, N2, F_ = _symmetry_shift_terms(res_two.x, xi_idx, norb)
+    new_two = two_electron - k2 * N2 - F_ / 4
+
+    params = np.hstack(([0.0, k2], xi[xi_idx]))
+    cost_func = partial(
+        _symmetry_shift_one_body_loss, one=one_electron, n_elec=n_elec, xi_idx=xi_idx
+    )  # Step 2: Reduce norm of one-body term
+
+    res_one = sp.optimize.minimize(cost_func, params, method=method)
+    k1, _, _, N1, _, _ = _symmetry_shift_terms(res_one.x, xi_idx, norb)
```

Comment:
Since it is a private function and we only use two of the returned terms, shall we make it return only those terms?

Reply:
Figured it out, nevermind.